app/code/Magento/Ui/view/base/web/js/grid/columns/image-preview.js

```diff
@@ -141,19 +156,42 @@ define([
             this._selectRow(record.rowNumber || null);
             this.visibleRecord(record._rowIndex);
 
-            img = $(this.previewImageSelector + ' img');
+            this.lastOpenedImage(record._rowIndex);
+            this.updateImageData();
+        },
 
-            if (img.get(0).complete) {
-                this.updateHeight();
-                this.scrollToPreview();
+        /**
+         * Update image data
+         */
+        updateImageData: function () {
+            var img = $(this.previewImageSelector + ' img');
+
+            if (!img.get(0)) {
+                setTimeout(function () {
+                    this.updateImageData();
+                }.bind(this), 100);
             } else {
-                img.load(function () {
+                if (img.get(0).complete) {
                     this.updateHeight();
                     this.scrollToPreview();
-                }.bind(this));
+                } else {
+                    img.load(function () {
+                        this.updateHeight();
+                        this.scrollToPreview();
+                    }.bind(this));
+                }
             }
+        },
 
-            this.lastOpenedImage(record._rowIndex);
+        /**
+         * Update displayed record
```

Comment:
It's better to make comments at least a bit more descriptive than method name if possible
```suggestion
         * Update preview displayed record data from the new items data if the preview is expanded
```